File: Binary Trees/Traversals/08-Iterative Preorder Traversal of Binary Tree.java
class Solution {
    public List<Integer> preorderTraversal(TreeNode root) {
        if(root == null){
            return new ArrayList<>();
        }

        Stack<TreeNode> st = new Stack<>();
        st.push(root);

        List<Integer> traversal = new ArrayList<>();

        while(!st.empty()){
            TreeNode temp = st.pop();

            traversal.add(temp.val);

            if(temp.right != null){
                st.push(temp.right);
            }
            if(temp.left != null){
                st.push(temp.left);
            }
        }

        return traversal;
    }
}

File: Binary Trees/Traversals/08-Iterative Preorder Traversal of Binary Tree.cpp
class Solution {
public:
    vector<int> preorderTraversal(TreeNode* root) {
        if(root == nullptr){
            return {};
        }

        stack<TreeNode*> st;
        st.push(root);

        vector<int> traversal;

        while(!st.empty()){
            TreeNode* temp = st.top();
            st.pop();

            traversal.push_back(temp -> val);

            if(temp -> right){
                st.push(temp -> right);
            }
            if(temp -> left){
                st.push(temp -> left);
            }
        }

        return traversal;
    }
};